oracle/src/sponge.rs

```diff
@@ -155,28 +155,24 @@ where
         let total_length = P::ScalarField::size_in_bits();
 
         x.iter().for_each(|x| {
-            // Big endian
-            let mut bits: Vec<bool> = x.into_repr().to_bits();
-            // Little endian
-            bits.reverse();
-            let mut bits: Vec<_> = (0..total_length)
+            // Padding
+            let bits: Vec<bool> = x.into_repr().to_bits_be();
```

Comment:
I think may be wrong

Reply:
Yeah the old code inserted the `0` values at the beginning of the `Vec`. Something like the following should work:
```rust
            let bits: Vec<bool> = x.into_repr().to_bits_be();
            let mut padding = vec![false, total_length - bits.len()]; // maybe `checked_sub`?
            padding.extend(bits);
			bits = padding;
```

Reply:
good catch! for some reason I remember simplifying this part but it looks like I forgot some code >.<

Reply:
fixed in https://github.com/o1-labs/marlin/pull/114/commits/25bcb80a24c10ebead8ed1cfa578450f9b07572c

Reply:
don't think your code would work @pratyush as total_length and bits.len() are not known at compile time

Reply:
The `vec![]` macro doesn't require compile-time lengths; run-time is fine =)

Reply:
woot! TIL :D